File: BTT4/BTT4 (BTT3 UPDATE)/BTT3/DisplayFriendListGUI.cpp
#include "DisplayFriendListGUI.h"
#include "Controller.h"
#include <iostream>

void DisplayFriendListGUI::RequestFriendList(string userID)
{
	Controller Controller_GUI;
	cout << "RequestFriendList" <<endl;
//	Controller_GUI.RequestFriendList(userID).unique();
	list<string> lst = Controller_GUI.RequestFriendList(userID);
	for (list<string>::iterator i = lst.begin(); i != lst.end(); ++i)
	cout  << "friend list (no sort): "<< *i <<  endl;
	cout << "if you want to see friend list for timw of sentting messege, press ctr+t or ctr+c to see for city !! " << endl;
	char key = getch();
	int value = key;
	if (value == 20) { 
		DisplayFriendListForTimeSentMess(userID);
	}
	if (value == 3) {
		DisplayFriendLisForCity(userID);
	}
}

void DisplayFriendListGUI::DisplayFriendListForTimeSentMess(string userID)
{
	Controller Controller_GUI;
	
	list<string> lst_temp1=Controller_GUI.TakeFriendListIDUserSentMess(userID);
	for (list<string>::iterator i = lst_temp1.begin(); i != lst_temp1.end(); ++i)
	{
		cout << *i << endl;
	//	string friendID(*i);
		
	}

}

void DisplayFriendListGUI::DisplayFriendLisForCity(string userID)
{
	Controller Controller_GUI;
	list<string> lst_temp1 = Controller_GUI.DisplayFriendLisForCity(userID);
	for (list<string>::iterator i = lst_temp1.begin(); i != lst_temp1.end(); ++i)
	{
		cout << *i << endl;
	}

}


DisplayFriendListGUI::DisplayFriendListGUI()
{
}


DisplayFriendListGUI::~DisplayFriendListGUI()
{
}

File: BTT4_PY1/BTT4_PY1/DisplayFriendListGUI.py
from Controller import Controller
from msvcrt import getch
class DisplayFriendListGUI(object):
	def RequestToDisplayFriendList(userID):
		controller = Controller
		print("friend list:")
		for row in controller.RequestToDisplayFriendList(userID):
			print (str(row))
		DisplayFriendListGUI.Takelist(userID)
		

	def Takelist(userID):
		controller = Controller
		print("press ctr+D to see defaul list or ctr+C see for city")
		input_key = ord(getch()) 
		if(input_key==4):
			print("defaul list")
			for row in controller.TakeFriendListSentMessToUser(userID):
				print(row)
			DisplayFriendListGUI.Takelist(userID)
		if(input_key==3):
			print("see for city")
			DisplayFriendListGUI.Takelist(userID)
			for row4 in controller.TakeCity(userID):
				print(row4)
			

